### ISSUE TITLE
ITerminalTransport missing functions for setting and resetting private modes

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2403]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#2403|
|Type|defect|
|Created|2007-01-15 23:14:14Z|

Private modes are useful for controlling wrapping behavior or hiding the cursor.  Most terminals support these.  ServerProtocol actually implements setPrivateModes and resetPrivateModes, but they should also be part of the interface and be provided by other implementations.

<details><summary>Searchable metadata</summary>

```
trac-id__2403 2403
type__defect defect
reporter__exarkun exarkun
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__conch conch
keywords__ 
time__1168902854000000 1168902854000000
changetime__1169140220000000 1169140220000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun removed owner|
|-|-|

Done in private-modes-2403

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_2"></a>|@therve set owner to @exarkun|
|-|-|

Some nitpicking:

 * unused string import in insults
 * in the reset method of TerminalBuffer, I'd use setPrivateModes to initialize the modes

When OK please merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun set status to `closed`|
|-|-|

(In [19393]) Merge private-modes-2403

Author: exarkun
Reviewer: therve
Fixes [#2403](https://github.com/twisted/twisted/issues/2403)

Add the DEC private mode manipulation methods which were previously implemented
by twisted.conch.insults.insults.ServerProtocol to ITerminalTransport and to
TerminalBuffer.  Add tests for these, as well as for their initial setting on
TerminalBuffer.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
